main.rb

```diff
@@ -1,33 +1,72 @@
-# frozen_string_literal: true
+require_relative 'app'
+require_relative 'book_refactor'
 
-require 'json'
-require_relative 'classes/author'
-require_relative 'classes/book'
-require_relative 'classes/games'
-require_relative 'classes/genre'
-require_relative 'classes/item'
-require_relative 'classes/label'
-require_relative 'classes/musicalbum'
-require_relative 'menu'
-require_relative 'actions'
+def main
+  puts 'Welcome to the Catalog of my things App!'
+  puts ' '
+  app = App.new
+  book_refactor = Refactor.new(app)
 
-# code to load data here
-# def load_data
-# end
+  loop do
+    display_options
+    input_number = gets.chomp.to_i
 
-# code to save data here
-# def save_data
-# end
+    if input_number == 10
+      puts "\e[31mThank you for using this app! ♥️ \e[0m"
+      break
+    end
+    if input_number < 4
+      handle_menu_choice(input_number, app, book_refactor)
+    elsif input_number > 3
+      actions(input_number, app)
+    end
+  end
+end
 
```

Comment:
- [ ] Kindly ensure the app is able to save data correctly. After deleting the `data` files that come with the app, any data I save cannot be retrieved when I restart the app. Kindly fix this minor issue.

Check the video below detailing the issue.
[cinnamon-2023-12-09T231717+0200.webm](https://github.com/BHObserver/Catalog-Of-My-Things/assets/81932478/ef9d166d-01dd-4e68-847f-ecebb91d9432)

#### Kindly note that this is happening for the `Add Book` and the `Add Music Album` options. Therefore the list options for the book and music are Not working. Kindly fix this. 
Also, there is a rewriting of the previously saved data once new data is added to the app. Check how the book that was saved before is overwritten by the new entry in the screencast below. kindly fix this as well
[cinnamon-2023-12-09T232617+0200.webm](https://github.com/BHObserver/Catalog-Of-My-Things/assets/81932478/348e19bf-d6ed-4a47-9827-268d70d9120d)